lib/Raven/Client.php

```diff
@@ -550,28 +538,25 @@ public static function getDefaultProcessors()
      * Sets the \Raven\Processor sub-classes to be used when data is processed before being
      * sent to Sentry.
      *
-     * @param $options
      * @return \Raven\Processor[]
      */
-    public function setProcessorsFromOptions($options)
+    public function createProcessors()
     {
         $processors = [];
-        foreach (\Raven\util::get($options, 'processors', static::getDefaultProcessors()) as $processor) {
-            /**
-             * @var \Raven\Processor        $new_processor
-             * @var \Raven\Processor|string $processor
-             */
-            $new_processor = new $processor($this);
+
+        foreach ($this->config->getProcessors() as $processor) {
+            /**  @var Processor $processorInstance */
+            $processorInstance = new $processor($this);
 
             if (isset($options['processorOptions']) && is_array($options['processorOptions'])) {
-                if (isset($options['processorOptions'][$processor])
-                    && method_exists($processor, 'setProcessorOptions')
-                ) {
-                    $new_processor->setProcessorOptions($options['processorOptions'][$processor]);
+                if (isset($options['processorOptions'][$processor]) && method_exists($processor, 'setProcessorOptions')) {
```

Comment:
>this check could be done with an interface

This is indeed one of the things I'm working on, I introduced a `ProcessorInterface` interface and this whole function will be refactored to support instantiation of the classes through a factory, but it's not the scope of this PR. The scope of this PR is to just extract the client options to their own class